src/goto-programs/goto_convert.cpp

```diff
@@ -1135,12 +1107,9 @@ void goto_convertt::convert_dowhile(
   goto_programt &dest,
   const irep_idt &mode)
 {
-  if(code.operands().size()!=2)
-  {
-    error().source_location=code.find_source_location();
-    error() << "dowhile takes two operands" << eom;
-    throw 0;
-  }
+  INVARIANT(
+    code.operands().size() == 2,
+    code.find_source_location().as_string() + ": dowhile takes two operands");
```

Comment:
This function should take a `code_dowhilet` instead of a `codet`, and then doesn't require this invariant here anymore.